pkg/kp/rcstore/consul_store.go

```diff
@@ -451,8 +451,32 @@ func (s *consulStore) SetDesiredReplicas(id fields.ID, n int) error {
 }
 
 func (s *consulStore) AddDesiredReplicas(id fields.ID, n int) error {
+	rcp, err := s.rcPath(id)
```

Comment:
instead of providing the logic here, maybe I could this facility available to all callers of `retryMutate`. The mutation function could take in two RCs - one is the RC fetched before any retries, and one is the RC most recently fetched. AddDesiredReplicas would use the original RC as it is using it here.